Sources/MessagingPush/MessagingPush+AppDelegate.swift

```diff
@@ -65,7 +65,27 @@ extension MessagingPushImplementation {
         switch response.actionIdentifier {
         case UNNotificationDefaultActionIdentifier: // push notification was touched.
             if let deepLinkurl = pushContent.deepLink {
-                UIApplication.shared.open(url: deepLinkurl)
+                // First, try to open the link inside of the host app. This is to keep compatability with Universal Links.
+                // Learn more of edge case: https://github.com/customerio/customerio-ios/issues/262
+                // Fallback to opening the URL system-wide if fail to open link in host app.
+                // Customers with Universal Links in their app will need to add this function to their `AppDelegate` which will get called with deep link:
+                // func application(_ application: UIApplication, continue userActivity: NSUserActivity, restorationHandler: @escaping ([UIUserActivityRestoring]?) -> Void) -> Bool
+
+                var didHostAppHandleLink = false
+                if deepLinkUtil.isLinkValidNSUserActivityLink(deepLinkurl) {
+                    logger.debug("Found a deep link inside of the push notification. Attempting to open deep link in host app, first.")
+
+                    let openLinkInHostAppActivity = NSUserActivity(activityType: NSUserActivityTypeBrowsingWeb)
+                    openLinkInHostAppActivity.webpageURL = deepLinkurl
+
+                    let didHostAppHandleLink = UIApplication.shared.delegate?.application?(UIApplication.shared, continue: openLinkInHostAppActivity, restorationHandler: { _ in }) ?? false
```

Comment:
Do we really want to redefine `didHostAppHandleLink` here? This means we're never changing the original `didHostAppHandleLink` from line 74, so it'll always be `false` when we check below on line 84.

Universal links are still not working for us, and based on our tests this is the culprit.